packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -133,61 +128,49 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [self.signIn signOut];
     result(nil);
   } else if ([call.method isEqualToString:@"disconnect"]) {
-    if ([self setAccountRequest:result]) {
-      [self.signIn disconnect];
-    }
+    [self.signIn disconnectWithCallback:^(NSError *error) {
+      [self respondWithAccount:@{} result:result error:nil];
+    }];
   } else if ([call.method isEqualToString:@"requestScopes"]) {
-    GIDGoogleUser *user = self.signIn.currentUser;
-    if (user == nil) {
-      result([FlutterError errorWithCode:@"sign_in_required"
-                                 message:@"No account to grant scopes."
-                                 details:nil]);
-      return;
+    id scopeArgument = call.arguments[@"scopes"];
+    if ([scopeArgument isKindOfClass:[NSArray class]]) {
+      self.requestedScopes = [self.requestedScopes setByAddingObjectsFromArray:scopeArgument];
     }
-
-    NSArray<NSString *> *currentScopes = self.signIn.scopes;
-    NSArray<NSString *> *scopes = call.arguments[@"scopes"];
-    NSArray<NSString *> *missingScopes = [scopes
-        filteredArrayUsingPredicate:[NSPredicate
-                                        predicateWithBlock:^BOOL(id scope, NSDictionary *bindings) {
-                                          return ![user.grantedScopes containsObject:scope];
-                                        }]];
-
-    if (!missingScopes || !missingScopes.count) {
```

Comment:
I didn't bother re-implementing checking if the scopes were already granted this since it's already handled by the API which returns `kGIDSignInErrorCodeScopesAlreadyGranted`.
https://github.com/google/GoogleSignIn-iOS/blob/20b21a1997abb13c9e9a7730461c8ae4cbaeb309/GoogleSignIn/Sources/GIDSignIn.m#L287-L292